shared/common-adapters/floating-menu.js

```diff
@@ -36,3 +37,28 @@ export default (props: Props) => {
     </FloatingBox>
   )
 }
+
+export type FloatingMenuParentProps = {
+  attachmentRef: ?React.Component<*, *>,
+  showingMenu: boolean,
+  setAttachmentRef: (?React.Component<*, *>) => void,
+  setShowingMenu: boolean => void,
+  toggleShowingMenu: () => void,
+}
+
+// TODO mock this for storybook w/ actions
+export const FloatingMenuParentHOC = compose(
```

Comment:
I think we shouldn't use recompose here. Even though `flow coverage` doesn't color `FloatingMenuParentHOC` red here, it seems to not know what its type is in places that import it; I get `(unknown)` using `flow type-at-pos` (via emacs). So I don't think it's worth the loss of types.

I think an approach similar to `HOCTimers` would work without too much more code, e.g.

```
type State = {
  attachmentRef: ...
  showingMenu: ...
}

const FloatingMenuParentHOC = <T: FloatingMenuParentProps>(
  ComposedComponent: React.ComponentType<T>
): React.ComponentType<$Diff<T, FloatingMenuParentProps>> => {
  class FloatingMenuParent extends React.Component<$Diff<T, FloatingMenuParentProps>, State> {
  ...
  // build an object with the setters in this._setters such that the type of this._setters & State == FloatingMenuParentProps.
  ...
  render() {
    return <ComposedComponent {...this.props} {...this._setters} {...this.state} />
  }
}
```